core/src/main/java/org/elasticsearch/cluster/ClusterName.java

```diff
@@ -33,6 +33,8 @@
     public static final Setting<ClusterName> CLUSTER_NAME_SETTING = new Setting<>("cluster.name", "elasticsearch", (s) -> {
         if (s.isEmpty()) {
             throw new IllegalArgumentException("[cluster.name] must not be empty");
+        } else if(s.contains(":")){
+            throw new IllegalArgumentException("[cluster.name] must not contains ':' character");
```

Comment:
1. There is a minor typo/grammatical mishap here - text should read "[cluster.name] must not _contain_ ':' character" 

2. Id consider putting this exception text into a final static variable somewhere it would make sense to put it. This text is currently used in two places in the code - once here, and once in a unit test - and the way things are now, if you want to change the contents of this text, you need to change two strings in two different places in the code. If you had this text in a final String variable, and you referenced that variable here and in the test, you would only need to change the string in one place.